lib/MusicBrainz/Server/Data/Artist.pm

```diff
@@ -465,12 +466,27 @@ sub _hash_to_row
 }
 
 sub load_related_info {
-    my ($self, @artists) = @_;
+    my ($self, $artists_ref, $user_lang) = @_;
 
     my $c = $self->c;
+    my @artists = @{$artists_ref};
     $c->model('ArtistType')->load(@artists);
     $c->model('Gender')->load(@artists);
     $c->model('Area')->load(@artists);
+
+    # Load and save aliases so Controller::Artist::show can use them later.
+    my $artist_aliases = $c->model('Artist')->alias->find_by_entity_ids(
+        map { $_->id } @artists,
+    );
+    for my $artist (@artists) {
+        my @aliases = @{ $artist_aliases->{$artist->id} };
+        $c->model('Artist')->alias_type->load(@aliases);
```

Comment:
Would it be better to make a single call that loads the alias types for all of the aliases instead of doing it once per artist here?

Reply:
> Would it be better to make a single call that loads the alias types for all of the aliases instead of doing it once per artist here?

Yes, it probably won't make a significant difference in production, but will where the PG connection latency is high (which happens when connecting via SSH tunnel, at least).

Reply:
Thanks, I've made this change for the next commit and it seems to work.